js/libs/keycloak-js/src/keycloak.js

```diff
@@ -1317,9 +1329,39 @@ function Keycloak (config) {
                     return createPromise().promise;
                 },
 
-                logout: function(options) {
-                    window.location.replace(kc.createLogoutUrl(options));
-                    return createPromise().promise;
+                logout: async function(options) {
+
+                    const logoutMethod = options?.logoutMethod ?? kc.logoutMethod;
+                    if (logoutMethod === "GET") {
+                        window.location.replace(kc.createLogoutUrl(options));
+                        return createPromise().promise;
```

Comment:
Since `logout()` is an async function we can return here directly without creating a Promise. Doing a return from an `async` function makes it's return type `Promise<void>` and resolves it immediately.
```suggestion
                        return;
```